src/consensus/aft/test/driver.h

```diff
@@ -151,6 +153,12 @@ class RaftDriver
 
     aft::ReplicatedDataType type = aft::ReplicatedDataType::raw;
     auto hooks = std::make_shared<kv::ConsensusHookPtrs>();
+    if (configuration.has_value() && retired_committed.has_value())
```

Comment:
Similar to `configuration`, `retired_committed` is a special replicate with a payload (a JSON config), but different semantics:

1. The configuration is kept in a local collection, subject to rollback
2. On compact(`idx`), set_retired_committed is called on the elements of configurations strictly older than `idx`.

This is, as far as I can tell, the smallest possible change necessary to emulate faithfully what the global commit hook does in https://github.com/microsoft/CCF/blob/0fa4538ef2bd86cba991175abdd9ce3c391294c2/src/node/node_state.h#L2147.